### PR TITLE
NEW: Includes debugBar (issue 9170)

### PR DESCRIPTION
**Pull request by rsanjoseo**
# Feature Request
It would be interesting to have a modern debugging tool in debug mode, such as:
https://github.com/maximebf/php-debugbar

## Use case
It shows a debug bar with information about execution, SQL queries, time consumed between strategic points and messages for debugging the code.
This bar would only be shown if LOG_DEBUG is greater than 0.

## Suggested implementation
Include the debugBar module through the composer. If LOG_DEBUG is greater than 0, place the JS and CSS code of that application on all pages, patch SQL queries and LOG messages to display them in the debug bar.

## Suggested steps
- Include degubBar in composer.
- Include the autoloader code.
- Modify the SQL queries to send the query to the SQL tab of the debug bar.
- Modify dol_syslog to send messages on the Messages tab of the debug bar.
- Include in the header and footer of each page the include of the debugBar js and css files

### CODE REVIEW

**Comment by rsanjoseo** [comment]
I do not find problems in PR.
To incorporate debugBar it has been necessary to make a composer update by including a new element.
I think the problems may come out there, but not because of the code introduced.
It's my first contribution and my english is pretty bad (sorry)

**Comment by frederic34** [comment]
you have used composer to install some library and you have updated many other like ckeditor, phpexcel etc...
please install manually only libraries needed

**Comment by eldy** [comment]
Your PR includes some external lib like "symphony" or "psr".
Does the added code really need this ?

**Comment by rsanjoseo** [comment]
It's true that it includes some symfony code. Little more than 700kb, which
some support bookstores like polyfill or yaml, but that according to
composer are necessary for debugbar operation.

On the other hand, the debugbar utility can be very useful to analyze the
execution time, the executed queries or show messages without breaking the
code.

File: htdocs/core/db/mssql.class.php [hunk 1]
```diff
@@ -324,9 +325,13 @@ function rollback($log='')
 	 */
 	function query($query,$usesavepoint=0,$type='auto')
 	{
-		$query = trim($query);
+        global $debugBar;
 
-		if (preg_match('/^--/',$query)) return true;
+        $query = trim($query);
+        if (isset($debugBar))
```
Comment (eldy) [inline htdocs/core/db/mssql.class.php:331]: You should not need to add code into each db drivers. Into this function query, like query of all db drivers, at end, there is already a call to 
dol_syslog
And this function include an architecture to allow to add as many logging output as we need.
You may have examples of log handlers into
core/modules/syslog
Each file mod_syslog_.... is a handler to manage a different output.

**Comment by eldy** [comment]
If composer add 700 KB of code just to add 1 log handler, it means you must not use composer and embbed the library you need and only the lib you need in a subdir without using composer.
Also, do you know there is already the ChromePHP and FirePHP log handler that seems to make a similar feature. php-debugbar may be more complete, but we must have control/knowledge of each necessary files and include only what we need.
For this composer is just a big mess, because it bring tons of dependencies we don't need and worse, it replace some version of some php lib with others non compatible... Just put the file you need into a subdir should be enough.

**Comment by librethic-code** [comment]
The big mess is we are not using Composer for whole project dependencies…

**Comment by librethic-code** [comment]
> we must have control/knowledge of each necessary files and include only what we need.

Composer allows this control of include files we need! By just reading `composer.json` file we have dependencies listing with upstream version. Obviously if we make changes into externals libs we should merge them into upstream repositories, that have not been correctly done for some time.

**Comment by rsanjoseo** [comment]
I recently used composer, but since I started using it I think it is an
essential tool.

**Comment by ksar-ksar** [comment]
I think that does not need to be in the core.
A module is better because it will be used only by developers.
And a module allready exist : https://www.dolistore.com/en/modules/996-Debug-bar.html

@+

**Comment by eldy** [comment]
Need is already addressed by the module debugbar
https://www.dolistore.com/en/modules/996-Debug-bar.html
so i close the PR

**Pull request closed without merging**